### ISSUE TITLE
[Bug] addPrefetchExcludes does not work

### ISSUE THREAD

**Issue by bradennapier** [react-static/react-static#1341]
### Environment

```
  System:
    OS: macOS Mojave 10.14.6
    CPU: (16) x64 Intel(R) Xeon(R) W-2140B CPU @ 3.20GHz
    Memory: 709.72 MB / 32.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.15.3 - ~/.nvm/versions/node/v10.15.3/bin/node
    Yarn: 1.15.2 - /usr/local/bin/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v10.15.3/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 78.0.3904.108
    Firefox: 66.0.1
    Safari: 12.1.2
  npmPackages:
    react-static: 7.2.3 => 7.2.3 
    react-static-plugin-preact: ^7.2.2 => 7.2.2 
    react-static-plugin-reach-router: ^7.2.3 => 7.2.3 
    react-static-plugin-sitemap: ^7.2.3 => 7.2.3 
    react-static-plugin-source-filesystem: ^7.2.3 => 7.2.3 
    react-static-plugin-styled-components: ^7.2.2 => 7.2.2 
```

### Steps to Reproduce the problem

It is quite possible I am doing this wrong in some way, but I followed the various pieces as closely as they provided information so not sure what i may be doing wrong here?

Following the reach-router guide and [addPrefetchExcludes](https://github.com/react-static/react-static/blob/master/docs/api.md#addprefetchexcludes) example

```
// App.jsx
import React from 'react';
import { Root, Routes, addPrefetchExcludes } from 'react-static';
import { Router } from '@reach/router';
import Home from './pages/index';

import './app.css';

// trying all types of options to make this work - nothing works :(
addPrefetchExcludes(['referral', '/referral', /referral/]);

function App() {
  return (
    <Root>
      <div className="content">
        <React.Suspense fallback={<div />}>
          <Router>
            <Home path="referral/:referralCode" />
            <Routes default />
          </Router>
        </React.Suspense>
      </div>
    </Root>
  );
}

export default App;
```

Now navigate to `referral/my-code` and it will produce a 404 error 

### Expected Behavior

Do not provide a 404 error as this breaks all meta tag handling -- in my case we simply want `referral/*` to render the same thing that `/` does to capture the URL path and show the same page

**Comment by bradennapier** [comment]
Alright so the solution to this was simple enough and was not the fault of the package -- but it is just confusing the way errors are produced

1. In development you do get a 404 error that does not actually seem to appear in production about getting `routeInfo` - so that request must not be checking the excludes value?
2. In production it does appear to work, however, i needed to add to my redirects for netlify:

```
/*    /index.html   200
```

**Comment by mamadOuologuem** [comment]
@bradennapier 
What about re-opening this issue? 
I think we shouldn't get a `404` even in a development env.

**Comment by SleeplessByte** [comment]
This is fixed in React Static 8.0+ but  we don't have the bandwith to backport it.
